### ISSUE TITLE
python frontend fails on windows

### ISSUE THREAD

**Issue by icemachined** [Fraunhofer-AISEC/cpg#1032]
# Stack trace

*Stack trace in case of an Exception*

jep.JepException: <class 'ModuleNotFoundError'>: No module named 'CPGPython'
	at D:\Workspace\luban\cpg\cpg-language-python\src\main\python\cpg.<module>(cpg.py:25)
	at jep.Jep.run(Native Method)
	at jep.Jep.runScript(Jep.java:247)
	at de.fraunhofer.aisec.cpg.frontends.python.PythonLanguageFrontend.parseInternal(PythonLanguageFrontend.kt:95)

# Unexpected behavior

When you setup CPG_JEP_LIBRARY on windows, system can't find CPGPython package

# Root cause
This is because 
```
val pyInitFile = classLoader.getResource("/CPGPython/__init__.py")
```
return path with '/' separator, but later
```
            var pyFolder = pyInitFile.file.dropLastWhile { it != File.separatorChar }
            pyFolder = pyFolder.dropLast(1)
            pyFolder = pyFolder.dropLastWhile { it != File.separatorChar }
            config.addIncludePaths(pyFolder)
```
code use system separator '\\' which is different from pyInitFile separators

You can fix it by using java.nio.file.Paths
```
config.addIncludePaths(Paths.get(pyInitFile.toURI()).parent.parent.toString())
```

Also in ```__init__.py``` unix separator is used
```
nsd_name = ".".join(self.fname.split("/")[-1].split(".")[:-1])
```

recommend to use library functions like this
```
nsd_name = ".".join(os.path.basename(self.fname).split(".")[:-1])
```

**Comment by icemachined** [comment]
Hi @maximiliankaul, i checked, this exception was fixed, 
but next exception was raised:
```
jep.JepException: <class 'RuntimeError'>: Illegal character in authority at index 7: file://src\test\resources\python\simple_class.py
	at D:\Workspace\luban\cpg\cpg-language-python\build\resources\main\CPGPython\_misc.add_mul_loc_infos(_misc.py:88)
	at D:\Workspace\luban\cpg\cpg-language-python\build\resources\main\CPGPython\_misc.add_loc_info(_misc.py:66)
	at D:\Workspace\luban\cpg\cpg-language-python\build\resources\main\CPGPython\_statements.handle_statement(_statements.py:40)
	at D:\Workspace\luban\cpg\cpg-language-python\build\resources\main\CPGPython\_statements.make_compound_statement(_statements.py:459)
	at D:\Workspace\luban\cpg\cpg-language-python\build\resources\main\CPGPython\_statements.handle_function_or_method(_statements.py:338)
	at D:\Workspace\luban\cpg\cpg-language-python\build\resources\main\CPGPython\_statements.handle_statement_impl(_statements.py:75)
	at D:\Workspace\luban\cpg\cpg-language-python\build\resources\main\CPGPython\_statements.handle_statement(_statements.py:39)
	at D:\Workspace\luban\cpg\cpg-language-python\build\resources\main\CPGPython\__init__.execute(__init__.py:84)
	at D:\Workspace\luban\cpg\cpg-language-python\src\main\python\cpg.parse_code(cpg.py:52)
	at D:\Workspace\luban\cpg\cpg-language-python\src\main\python\cpg.parse_code(cpg.py:58)
	at jep.Jep.invoke(Native Method)
	at jep.Jep.invoke(Jep.java:259)
	at de.fraunhofer.aisec.cpg.frontends.python.PythonLanguageFrontend.parseInternal(PythonLanguageFrontend.kt:70)
	at de.fraunhofer.aisec.cpg.frontends.python.PythonLanguageFrontend.parse(PythonLanguageFrontend.kt:47)
	at de.fraunhofer.aisec.cpg.TranslationManager.parse(TranslationManager.kt:359)
	at de.fraunhofer.aisec.cpg.TranslationManager.parseParallel$lambda$10(TranslationManager.kt:272)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.exec(CompletableFuture.java:1760)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.helpAsyncBlocker(ForkJoinPool.java:1253)
	at java.base/java.util.concurrent.ForkJoinPool.helpAsyncBlocker(ForkJoinPool.java:2237)
	at java.base/java.util.concurrent.CompletableFuture.waitingGet(CompletableFuture.java:1887)
	at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2072)
	at de.fraunhofer.aisec.cpg.TranslationManager.parseParallel(TranslationManager.kt:284)
	at de.fraunhofer.aisec.cpg.TranslationManager.runFrontends(TranslationManager.kt:228)
	at de.fraunhofer.aisec.cpg.TranslationManager.analyze$lambda$2(TranslationManager.kt:92)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.exec(CompletableFuture.java:1760)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
Caused by: java.net.URISyntaxException: Illegal character in authority at index 7: file://src\test\resources\python\simple_class.py
	at java.base/java.net.URI$Parser.fail(URI.java:2974)
	at java.base/java.net.URI$Parser.parseAuthority(URI.java:3308)
	at java.base/java.net.URI$Parser.parseHierarchical(URI.java:3219)
	at java.base/java.net.URI$Parser.parse(URI.java:3175)
	at java.base/java.net.URI.<init>(URI.java:623)
	... 23 more
```

this can be fixed by using
`File(self.fname).getCanonicalPath()`

instead of `URI("file://" + self.fname)` in _misc.py

**Comment by icemachined** [comment]
@maximiliankaul  also replacing 
`URI("file://" + self.fname)`
to
`File(self.fname).toURI()`
fixes this issue. because it converts file path to unix style

**Comment by maximiliankaul** [comment]
> @maximiliankaul also replacing `URI("file://" + self.fname)` to `File(self.fname).toURI()` fixes this issue. because it converts file path to unix style

I think the issue was caused by using relative paths for URIs. This is fixed in the current #1053. Are you still having problems with PR on Windows?

**Comment by icemachined** [comment]
> > @maximiliankaul also replacing `URI("file://" + self.fname)` to `File(self.fname).toURI()` fixes this issue. because it converts file path to unix style
> 
> I think the issue was caused by using relative paths for URIs. This is fixed in the current #1053. Are you still having problems with PR on Windows?

@maximiliankaul yes, the problem still persists on my windows machine with updated pr because URI parser doesn't expect windows file separator in URI string and throws this exception. File.toURI normalize file path , it gets AbsolutePath and replaces windows separators to unix like '/'.

**Comment by icemachined** [comment]
@maximiliankaul checked last update, everything is fine, all python tests have passed 👍
